### ISSUE TITLE
Escape characters in doctests are treated incorrectly with Python 2

### ISSUE THREAD

**Issue by erosennin** [sphinx-doc/sphinx#2062]
Recent Git versions of Sphinx are making some doctests fail because escape characters in the doctest output are decoded instead of being interpreted verbatim.

Here is a minimal example.

`index.rst`:

``` rst
.. doctest::
    >>> print('one\ntwo')
    one
    two
    >>> print(r'one\ntwo')
    one\ntwo
```

`conf.py`:

``` python
extensions = [
    'sphinx.ext.doctest',
]
master_doc = 'index'
```

The test fails with Python 2:

```
$ sphinx-build -b doctest . build/doctest
Running Sphinx v1.4a0+
loading pickled environment... done
building [mo]: targets for 0 po files that are out of date
building [doctest]: targets for 1 source files that are out of date
updating environment: 0 added, 0 changed, 0 removed
looking for now-outdated files... none found
running tests...

Document: index
---------------
**********************************************************************
File "index.rst", line 5, in default
Failed example:
    print(r'one\ntwo')
Expected:
    one\ntwo
Got:
    one
    two
**********************************************************************
1 items had failures:
   1 of   2 in default
2 tests in 1 items.
1 passed and 1 failed.
***Test Failed*** 1 failures.

Doctest summary
===============
    2 tests
    1 failure in tests
    0 failures in setup code
    0 failures in cleanup code
build finished with problems.
```

The same test passes with Python 3.

**Comment by erosennin** [comment]
Seems to be introduced by 82e05e7a9c7865c5f609e31ba70bab02ab521616.

**Comment by shimizukawa** [comment]
Fixed at e85ea7e. Thanks for your reporting!
